### PR TITLE
Allow prereleases from prepare-release workflow

### LINKED ISSUE #7551: Update release-on-comment script to handle release candidates
As discussed in https://github.com/pytest-dev/pytest/pull/7548#issuecomment-664988552

### PR DESCRIPTION
**Pull request by nicoddemus**
Unfortunately untested, as it is only available once it lands on `main`.

Fix #7551

### CODE REVIEW

File: .github/workflows/prepare-release-pr.yml [hunk 1]
```diff
@@ -11,6 +11,10 @@ on:
         description: 'Major release? (yes/no)'
         required: true
         default: 'no'
+      prerelease:
+        description: 'Prerelease (ex: "rc1", including the quotes)'
+        required: true
+        default: '""'
```
Comment (asottile) [inline .github/workflows/prepare-release-pr.yml:17]: the quoting on this seems weird -- I would expect `default: ''` to be better (coupled with a suggested change below)

File: .github/workflows/prepare-release-pr.yml [hunk 2]
```diff
@@ -34,9 +38,9 @@ jobs:
     - name: Prepare release PR (minor/patch release)
       if: github.event.inputs.major == 'no'
       run: |
-        tox -e prepare-release-pr -- ${{ github.event.inputs.branch }} ${{ secrets.chatops }}
+        tox -e prepare-release-pr -- ${{ github.event.inputs.branch }} ${{ secrets.chatops }} --prerelease ${{ github.event.inputs.prerelease }}
```
Comment (asottile) [inline .github/workflows/prepare-release-pr.yml:41]: and `--prerelease '${{ github.event.inputs.prerelease }}'`

**Comment by The-Compiler** [comment]
@nicoddemus What are your plans on this? If we want a 7.0.0rc1 (or 6.3.0rc1), we should probably get this in first.

**Comment by nicoddemus** [comment]
Argh completely missed this one.

I'm not sure I will be able to get to this today (I'll do my best), but if anybody wants to take this please go ahead. 👍

File: .github/workflows/prepare-release-pr.yml (hunk 1 above)
Comment (nicoddemus) [reply .github/workflows/prepare-release-pr.yml:17]: Done, thanks!

File: .github/workflows/prepare-release-pr.yml (hunk 2 above)
Comment (nicoddemus) [reply .github/workflows/prepare-release-pr.yml]: Done, thanks!

**Comment by nicoddemus** [comment]
Applied @asottile's suggestions. 👍

File: .github/workflows/prepare-release-pr.yml [hunk 3]
```diff
@@ -11,6 +11,10 @@ on:
         description: 'Major release? (yes/no)'
         required: true
         default: 'no'
+      prerelease:
+        description: 'Prerelease (ex: "rc1", including the quotes)'
```
Comment (The-Compiler) [inline .github/workflows/prepare-release-pr.yml:15]: Is the "including the quotes" still accurate now?

Comment (nicoddemus) [reply .github/workflows/prepare-release-pr.yml:15]: Nope, thanks for catching it. 👍

Comment (nicoddemus) [reply .github/workflows/prepare-release-pr.yml]: Fixed!

**Review by The-Compiler — APPROVED** [review]
LGTM now! Fingers crossed that it won't fail in some way which is totally obvious in hindsight but we all missed somehow :see_no_evil:

**Comment by nicoddemus** [comment]
> Fingers crossed that it won't fail in some way which is totally obvious in hindsight but we all missed somehow

Heh ditto! 😁

**Comment by The-Compiler** [comment]
Hmm wait, I think this is only a part of the equation, though? At least according to [RELEASING.rst](https://github.com/pytest-dev/pytest/blob/main/RELEASING.rst) we're still triggering @pytestbot via a release comment (rather than triggering the workflow manually), so [release-on-comment.py](https://github.com/pytest-dev/pytest/blob/main/scripts/release-on-comment.py) would need an update to support something like a `@pytestbot please prepare [major] prerelease from ...` too, no?

**Comment by nicoddemus** [comment]
> t least according to RELEASING.rst we're still triggering @pytestbot via a release comment (rather than triggering the workflow manually)

Those are not up to date, we have made the last two releases (or so) with this new workflow, I think we can remove the `release-on-comment` one as obsolete. 👍

**Pull request merged**